taxonomy-editor-frontend/src/pages/project/search/index.tsx

```diff
@@ -1,99 +1,77 @@
+  const [currentPage, setCurrentPage] = useState<number>(
+    parseInt(pageParam ?? "1")
+  );
 
-const SearchNode = ({ taxonomyName, branchName }: SearchNodeProps) => {
-  const [searchInput, setSearchInput] = useState("");
-  const [queryFetchString, setQueryFetchString] = useState("");
+  const { data: entryNodeSearchResult } = useQuery({
+    queryKey: [
+      "searchEntryNodesTaxonomyNameBranchNodesEntryGet",
+      branchName,
+      taxonomyName,
+      q,
+      currentPage,
+    ],
+    queryFn: async () => {
+      const nodesResult =
+        await DefaultService.searchEntryNodesTaxonomyNameBranchNodesEntryGet(
+          branchName ?? "",
+          taxonomyName ?? "",
+          q,
+          currentPage
+        );
+      setSearchExpression(nodesResult.q);
+      setSearchParams((prevSearchParams) => ({
+        ...prevSearchParams,
+        q: nodesResult.q,
+      }));
+      return nodesResult;
+    },
+  });
 
   return (
-    <Box>
-      <Grid
-        container
-        display="flex"
-        flexDirection="column"
-        alignItems="center"
-        justifyContent="flex-start"
-        gap="20px"
-      >
-        <Typography variant="h3">Search</Typography>
-        <Box
-          component="img"
-          width={100}
-          height={100}
-          src={classificationImgUrl}
-          alt="Classification Logo"
-        />
-        <form
-          onSubmit={(event) => {
-            event.preventDefault();
-            setQueryFetchString(searchInput.trim());
-          }}
-        >
-          <TextField
-            sx={{ width: 350 }}
-            InputProps={{
-              endAdornment: (
-                <InputAdornment position="end">
-                  <IconButton
-                    disabled={searchInput.trim().length === 0}
-                    type="submit"
-                  >
-                    <SearchIcon />
-                  </IconButton>
-                </InputAdornment>
-              ),
-            }}
-            onKeyDown={(e) => {
-              if (e.keyCode === ENTER_KEYCODE && searchInput.length !== 0) {
-                setQueryFetchString(searchInput.trim());
-              }
-            }}
-            onChange={(event) => {
-              setSearchInput(event.target.value);
-            }}
-            value={searchInput}
-          />
-        </form>
-      </Grid>
-      {queryFetchString !== "" && (
-        <SearchResults
-          query={queryFetchString}
-          taxonomyName={taxonomyName}
-          branchName={branchName}
-        />
-      )}
+    <Box sx={{ display: "flex", flexWrap: "wrap" }}>
+      <SearchExpressionInput
+        searchExpression={searchExpression}
+        setSearchExpression={setSearchExpression}
+        setCurrentPage={setCurrentPage}
+        setQ={setQ}
+      />
+      <FiltersArea
+        setCurrentPage={setCurrentPage}
+        setQ={setQ}
+        filters={entryNodeSearchResult?.filters ?? []}
```

Comment:
This creates weird states: when you go and fetch a new search result, `entryNodeSearchResult` is set to `undefined` so the filters are reset during that time
You may want to use `placeholderData` and `isPlaceholderData` for this: https://tanstack.com/query/latest/docs/framework/react/guides/migrating-to-v5#removed-keeppreviousdata-in-favor-of-placeholderdata-identity-function